TwipsterApp/TwipsterMainWindow.xaml.cs

```diff
@@ -37,9 +52,39 @@ private void OnLogOutButtonClicked(object sender, RoutedEventArgs e)
             Close();
         }
 
+        private void OnShowAllPostsButtonClicked(object sender, RoutedEventArgs e)
+        {
+            using var context = new TwipsterDbContext();
+            PostGridRefresh(context);
+        }
+
+        private void OnShowSelectedUserPostsClicked(object sender, RoutedEventArgs e)
+        {
+            using var context = new TwipsterDbContext();
+            var selectedUserLogin = UsersGrid.SelectedItem as SelectedUserModel;
+            var selectedUser = context.Users.Single(x => x.Login == selectedUserLogin.Login);
+            var userPostsList = context.Posts.Where(x => x.UserId == selectedUser.Id).ToList();
+            PostsGrid.ItemsSource = userPostsList;
+        }
+
         private void PutCurrentUserInformationToTextBoxt(User currentUser)
         {
-            CurrentUserTextBlock.Text = $"{currentUser.Name} {currentUser.Surname} \nDate of birth: {currentUser.BirthDate.Date.ToString()} \nLogin: {currentUser.Login}";
+            CurrentUserTextBlock.Text = $"{currentUser.Name} {currentUser.Surname} \nDate of birth: {currentUser.BirthDate.Date} \nLogin: {currentUser.Login}";
         }
+
+        public void PostGridRefresh(TwipsterDbContext context)
+        {
+            var postsList = context.Posts.OrderBy(x => x.PostTime)
+                                .Select(x => new PostViewModel
+                                {
+                                    Name = x.User.Name,
```

Comment:
Did you check? Does it work?

Reply:
True, had rework "OnShowSelectedUserPostsClicked" func.